docs/how_to/query.md

```diff
@@ -51,6 +51,25 @@ You will notice that in the setup linked above, M3DB has just one unaggregated n
     all: false
 ```
 
+## ID generation
+
+The default generation scheme for IDs is unfortunately prone to collisions, but is left as is for back-compat reasons. It is suggested to set the ID generation scheme to one of either "quoted" or "prepend_meta". Quoted generation scheme yields the most human-readable IDs, whereas Prepend_Meta is better for more compact IDS, or if tags are expected to contain non-ASCII characters. To set the ID generation scheme, add the following to your coordinator configuration yaml file:
```

Comment:
What happens if we change the default to not be fucked? What kind of issues could it cause?

Reply:
So that would be the ideal case, unfortunately any current users (especially those with a ton of series) will see double incoming metrics

Reply:
Feels wrong to have the default be fucked for b/w compat. Would be better to break compat but expect users to change the default to non-sane because we'll publish guides/release notes/etc. 

Or we'll always be stuck in this place.

Reply:
Honestly, the default case is not *too* bad unless there's weird cases that collide like: 

`[{t1:v1},{t2:v2}]` -> t1=v1,t2=v2,
`[{t1:v1,t2:v2}]`   -> t1=v1,t2=v2,

I doubt users are jamming `,` and `=` characters into their tag name/values, but the alternative generation schemes will cover that case

Reply:
I agree it's kinda lame at the moment; I guess with the number of users we currently have, it's still manageable to spread awareness that their metric count will double and let them know they can explicitly set the default case to `legacy` if that presents an untenable situation

Reply:
> it's still manageable to spread awareness that their metric count will double and let them know they can explicitly set the default case to `legacy` if that presents an untenable situation

Yeah, if we're going to bite that bullet, it's not going to get better if we wait. Could we provide any data rewrite tooling to upgrade folks' existing data to the new scheme? Not sure if it's worth it (and/or if we have a good way to do that--map/reduce style primitives would be nice here), but that would be the way to go if we wanted to ease the pain a bit. Then again, I'm assuming any data rewrite would probably imply either downtime or some tricky code on our part to avoid downtime so... maybe best to just have double the metrics for a bit

Reply:
We decided to go a way; explicitly setting scheme in config is now necessary in the new version, with no default values. If users are happy to take on the cost of doubled cardinality, they can elect to choose a different scheme than `legacy`.

After some time we'll relax the condition that scheme is specified, and default to `quoted` (probably). This way, current users who do not want to take on the extra cardinality are able to set their configs correctly for any future updates.